### PR TITLE
Welcome Toast for unregistered Users

### LINKED ISSUE #5425: Info block on wk in public datasets
## Detailed Description
Add an informational block about the tool webKnossos to the view mode of a public dataset. It should only be shown when viewing a published dataset and not logged in.

## Context
webKnossos is used as dataset hosting for several publications. This is a great opportunity to explain how webKnossos can be used for similar research.

### PR DESCRIPTION
**Pull request by philippotto**
### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- open the a public tracing (or dataset) while being logged out
- --> welcome toast should appear
- reload --> toast should not appear
- perform `localStorage.clear()` and reload again to see the toast once more

### Issues:
- fixes #5425 

------
- [ ] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Updated [(unreleased) migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes
- [ ] Needs datastore update after deployment
- [x] Ready for review

### CODE REVIEW

File: .circleci/not-on-master.sh [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #!/usr/bin/env bash
 set -Eeuo pipefail
 
-if [ "${CIRCLE_BRANCH}" == "master" ]; then
-  echo "Skipping this step on master..."
-else
-  exec "$@"
-fi
+# if [ "${CIRCLE_BRANCH}" == "master" ]; then
```
Comment (philippotto) [inline .circleci/not-on-master.sh:4]: revert before merging

File: conf/application.conf [hunk 2]
```diff
@@ -72,7 +72,7 @@ features {
   discussionBoardRequiresAdmin = false
   addForeignDataset = false
   hideNavbarLogin = false
-  isDemoInstance = false
+  isDemoInstance = true
```
Comment (philippotto) [inline conf/application.conf:75]: revert before merging

File: frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  });
+}
+
+export default function WelcomeToast() {
+  const activeUser = useSelector(state => state.activeUser);
+
+  useEffect(() => {
+    // if (features().isDemoInstance) {
+    //   return;
+    // }
```
Comment (philippotto) [inline frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js:57]: comment in before merging

File: frontend/stylesheets/main.less [hunk 4]
```diff
@@ -367,3 +367,62 @@ img.keyboard-mouse-icon:first-child {
 .transparent-background-on-hover:hover {
   background: transparent;
 }
+
+.webknossos-welcome-toast {
+  background: url(https://res2.weblium.site/res/5ffc7e249a9fc00021e2921b/60a274863cb13f0021eb0e7a_optimized_3840.webp)
```
Comment (philippotto) [inline frontend/stylesheets/main.less:372]: todo: add to repo (in small size)

File: frontend/javascripts/oxalis/view/action-bar/tracing_actions_view.js [hunk 5]
```diff
@@ -397,16 +397,23 @@ class TracingActionsView extends React.PureComponent<Props, State> {
           ),
         ]
       : [
-          <ButtonComponent key="read-only-button" type="danger" disabled>
+          <ButtonComponent
+            key="read-only-button"
+            type="danger"
+            disabled
+            style={{ backgroundColor: "var(--ant-warning-dark-5)" }}
```
Comment (philippotto) [inline frontend/javascripts/oxalis/view/action-bar/tracing_actions_view.js:404]: I changed the bg color of the read-only button from red to orange (as it's been before the antd update).

File: frontend/javascripts/oxalis/view/action-bar/tracing_actions_view.js [hunk 6]
```diff
@@ -397,16 +397,23 @@ class TracingActionsView extends React.PureComponent<Props, State> {
...
-            Copy To My Account
-          </AsyncButton>,
+          activeUser != null ? (
+            <AsyncButton
+              key="copy-button"
+              icon={<FileAddOutlined />}
+              onClick={this.handleCopyToAccount}
+            >
+              Copy To My Account
+            </AsyncButton>
```
Comment (philippotto) [inline frontend/javascripts/oxalis/view/action-bar/tracing_actions_view.js:415]: "copy to my account" was visible when being logged out.

File: frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+      <div>
+        <span className="logo" />
+        welcome to webKnossos!
+      </div>
+    ),
+    description: (
+      <div>
+        You are viewing this dataset on webKnossos. webKnossos is an online tool to visualise and
+        collaboratively annotate large image datasets. Try out the annotation features on the public
+        dataset below. If this is for you, sign up for free, upload your data and get started!
```
Comment (philippotto) [inline frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js:19]: @hotzenklotz any feedback for the final wording?

File: frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+import { Button, notification } from "antd";
+import { useSelector } from "react-redux";
+import features from "features";
+
+function showWelcomeToast() {
+  notification.open({
+    message: (
+      <div>
+        <span className="logo" />
+        welcome to webKnossos!
```
Comment (normanrz) [inline frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js:12]:
```suggestion
        Welcome to webKnossos!
```

File: frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+        dataset below. If this is for you, sign up for free, upload your data and get started!
+        <div style={{ marginTop: 12, marginLeft: 200 }}>
+          <Button
+            type="primary"
+            href="/auth/login"
+            target="_blank"
+            rel="noopener noreferrer"
+            style={{ background: "#256ff7" }}
+          >
+            Sign Up
```
Comment (normanrz) [inline frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js:28]:
```suggestion
            Create a free account
```

File: frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+    ),
+    description: (
+      <div>
+        You are viewing this dataset on webKnossos. webKnossos is an online tool to visualise and
+        collaboratively annotate large image datasets. Try out the annotation features on the public
+        dataset below. If this is for you, sign up for free, upload your data and get started!
+        <div style={{ marginTop: 12, marginLeft: 200 }}>
+          <Button
+            type="primary"
+            href="/auth/login"
```
Comment (normanrz) [inline frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js:23]:
```suggestion
            href="/auth/signup"
```

File: frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js (hunk 7 above)
Comment (normanrz) [reply frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js:19]: webKnossos is a web-based platform for visualization, annotation, and sharing of large-scale 3D image datasets. Try out the annotation features and upload your own data with a free account.

Comment (hotzenklotz) [reply frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js:19]: see norman's suggestion below, please

File: frontend/stylesheets/main.less (hunk 4 above)
Comment (philippotto) [reply frontend/stylesheets/main.less]: done

**Comment by philippotto** [comment]
@normanrz @hotzenklotz I incorporated all your feedback. Feel free to check out the newest version here: https://infobanner.webknossos.xyz/datasets/sample_organization/dsA/view?token=axiBBcGKhoeuZuz2mPsADw#108,926,1024,0,1.000

once again: `localStorage.clear()` might be necessary to see the toast. if there are no vetos, I would pass this along to actual code review.

File: frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+        <span className="logo" />
+        Welcome to webKnossos!
+      </div>
+    ),
+    description: (
+      <div>
+        webKnossos is a web-based platform for visualization, annotation, and sharing of large-scale
+        3D image datasets. Try out the annotation features and upload your own data with a free
+        account.
+        <div style={{ marginTop: 12, display: "flex", justifyContent: "center" }}>
```
Comment (normanrz) [inline frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js:20]: I think left-aligned looks better 
Screenshot 2021-06-28 at 15 02 05

Also, the blue looks slightly different than the blue in the "create annotation" button. Is that something you could re-use from ant css classes?

Comment (philippotto) [reply frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js]: two very good points! wasn't sure about the left aligned version first, which is why I centered it, but now I agree with you. the special blue color was the consequence of some trouble with the antd "primary" style not working at first, which I forgot about. Both issues should be fixed now:

image

**Comment by philippotto** [comment]
@daniel-wer could you review the front-end changes? feel free to check out the dev instance, too :) afterwards, I would uncomment the necessary code (which makes testing on the dev instance impossible) and re-enable the important CI steps.

File: frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js (hunk 3 above)
Comment (daniel-wer) [reply frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js:51]: Is this condition correct or does it need to be negated? I would've thought this to be especially relevant for the demo instance.

**Review by daniel-wer — APPROVED** [review]
Very nice, I dig the design :)

File: frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js (hunk 3 above)
Comment (philippotto) [reply frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js:51]: you're right, this needs to be inverted!

Comment (hotzenklotz) [reply frontend/javascripts/oxalis/view/novel_user_experiences/welcome_toast.js]: good catch!

**Pull request merged**